### ISSUE TITLE
AVS support

### ISSUE THREAD

**Issue by WeirdTreeThing** [jmontleon/pixelbook-fedora#67]
Hello, 
You had made [this issue](https://github.com/eupnea-linux/audio-scripts/issues/39) on my old chromebook audio repo (new one [here](https://github.com/WeirdTreeThing/chromebook-linux-audio)) about using the AVS driver on eve. At the time only the speakers and hdmi were supported, but now the headphone codec should also be supported. I'd like to add support to my script but need someone to test it. If you want to try it you will need to build a kernel with [this](https://patchwork.kernel.org/project/alsa-devel/patch/Y6No3WsiZ7Sbg35u@tora/) patch and put [this](https://github.com/WeirdTreeThing/chromebook-linux-audio/blob/rt5663/conf/avs/tplg/rt5663-tplg.bin) topology in `/lib/firmware/intel/avs/`. If it works feel free to use that topology in your repo, I'm sure working with AVS will be a lot easier than the old skl SST driver. Mic still wont work though.

**Comment by jmontleon** [comment]
Thanks for the info! I will definitely give it a shot sometime this week!

**Comment by jmontleon** [comment]
Sorry about the delay. I put the topology file in place and have a kernel building with the patch. I should be able to give some feedback tomorrow.

**Comment by jmontleon** [comment]
Should I need `max98927-tplg.bin` and `hda-8086280b-tplg.bin` as well? I saw logs in dmesg complaining about `max98927-tplg.bin` missing and put it in place and now it's saying:
```
[    6.710297] snd_soc_avs 0000:00:1f.3: Direct firmware load for intel/avs/max98927-tplg.bin failed with error -2
[    6.710307] snd_soc_avs 0000:00:1f.3: request topology "intel/avs/max98927-tplg.bin" failed: -2
[    6.710311] snd_soc_avs 0000:00:1f.3: ASoC: error at snd_soc_component_probe on avs_max98927.1-platform: -2
[    6.710326] avs_max98927 avs_max98927.1: ASoC: failed to instantiate card -2
```
and this one, I have no idea where to get if needed. I see the `a` but not `b` in your repo.
```
[    6.756079] snd_soc_avs 0000:00:1f.3: Direct firmware load for intel/avs/hda-8086280b-tplg.bin failed with error -2
[    6.756085] snd_soc_avs 0000:00:1f.3: request topology "intel/avs/hda-8086280b-tplg.bin" failed: -2
[    6.756092] snd_soc_avs 0000:00:1f.3: ASoC: error at snd_soc_component_probe on hdaudioB0D2-platform: -2
[    6.756107] avs_hdaudio avs_hdaudio.2: ASoC: failed to instantiate card -2
```

I did move `rt5663-tplg.bin` out of the way and saw errors about it failing to load, moved it back again, and they went away, so that one does seem to be loading OK. 

I also saw:
```
[    6.761729] avs_rt5663 avs_rt5663.2: ASoC: Parent card not yet available, widget card binding deferred
```

`aplay -l` just shows
```
$ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 1: avsrt5663 [avs_rt5663], device 1: Headset (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

Full dmesg:
[dmesg.txt](https://github.com/jmontleon/pixelbook-fedora/files/11712521/dmesg.txt)

Also, I built a patched 6.4.0-5. If I should be patching a different version let me know and I'll rebuild.

**Comment by WeirdTreeThing** [comment]
> Should I need max98927-tplg.bin and hda-8086280b-tplg.bin as well? I saw logs in dmesg complaining about max98927-tplg.bin missing and put it in place and now it's saying:

Yes you do need those for HDMI audio and speakers.

> and this one, I have no idea where to get if needed. I see the a but not b in your repo.

I need to add more HDMI configurations to my repo, I will get this done soon.

> [    6.761729] avs_rt5663 avs_rt5663.2: ASoC: Parent card not yet available, widget card binding deferred

No need to worry about that.

> $ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 1: avsrt5663 [avs_rt5663], device 1: Headset (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0

Looks good

Now to create UCM, I will need a dump of all the controls. To get this, run `amixer -c<cardid> contents > rt5663-contents.txt` and upload that file here. Replace <cardid> with the id of rt5663, for example, `card 1: avsrt5663 [avs_rt5663], device 1: Headset (*) []` shows the cardid is 1.

> Also, I built a patched 6.4.0-5. If I should be patching a different version let me know and I'll rebuild.

That works fine.

**Comment by WeirdTreeThing** [comment]
As for the error loading max98927, the way I have things working currently is with using DSM instead of peakvol. DSM requires a ChromeOS 5.10 kernel while peakvol works on mainline linux. I'm going to switch my repo to use peakvol for max98927/max98373 eventually, for now I can create a branch which has the peakvol topologies.

**Comment by WeirdTreeThing** [comment]
The HDMI configuration you need can be found [here](https://github.com/WeirdTreeThing/chromebook-linux-audio/blob/main/conf/avs/tplg/hda-8086280b-tplg.bin) and the peakvol tplg can be found [here](https://github.com/WeirdTreeThing/chromebook-linux-audio/blob/peakvol/conf/avs/tplg/max98927-tplg.bin).

**Comment by jmontleon** [comment]
Cool! I'll give these a try tonight and post the output you requested too! Than you!

**Comment by jmontleon** [comment]
The firmwares all look to be loading. 
```
$ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: avsmax98927 [avs_max98927], device 1: Built-in Speakers (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: hdaudioB0D2 [hdaudioB0D2], device 1: HDMI1 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: hdaudioB0D2 [hdaudioB0D2], device 2: HDMI2 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: hdaudioB0D2 [hdaudioB0D2], device 3: HDMI3 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 3: avsrt5663 [avs_rt5663], device 1: Headset (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

Output of `dmesg` and `amixer -c3 contents > rt5663-contents.txt` attached.
[dmesg.txt](https://github.com/jmontleon/pixelbook-fedora/files/11713542/dmesg.txt)
[rt5663-contents.txt](https://github.com/jmontleon/pixelbook-fedora/files/11713543/rt5663-contents.txt)

**Comment by WeirdTreeThing** [comment]
UCM for rt5663 can be found [here](https://github.com/WeirdTreeThing/chromebook-ucm-conf/tree/main/avs/avs_rt5663). Keep in mind the the exact mixer values might not be correct and the pcm for the headset mic might also not be correct. You can open alsamixer, select rt5663, and change some of the values around until it all sounds right, then dump the mixer values again. Also run `arecord -l` to list all the recording pcms.

**Comment by jmontleon** [comment]
I haven't been able to get any audio to play out of the speakers or headphones. Playing in alsamixer I can get some white noise out of the headphones, but that's all I've been able to manage so far. I haven't tried HDMI. 

Both Headphones and Speakers show up as `Pro Audio` in pavucontrol and what I assume is the three HDMI show up as `Pro Audio` `Pro Audio 2` and `Pro Audio 3`, so it's difficult to tell which is which. I removed the other two topology files to find the one for the headphones.

I can keep playing around and see if I can get anything out of it.

**Comment by WeirdTreeThing** [comment]
Looks like an issue with UCM. If you are using pulseaudio, run `systemctl --user stop pulseaudio.socket` and then run `pulseaudio -v > pulseaudio.log 2>&1`  then Ctrl + C after a few seconds. If you are using pipewire, run `spa-acp-tool -c <CardId> -vvvv info > pipewire-<CardId>.log 2>&1`, do this for each card listed in `cat /proc/asound/cards`.

**Comment by jmontleon** [comment]
Attached for each of these:
```
$ cat /proc/asound/cards
 0 [hdaudioB0D2    ]: hdaudioB0D2 - hdaudioB0D2
                      Google-Eve-1.0
 1 [avsprobemb     ]: avs_probe_mb - avs_probe_mb
                      Google-Eve-1.0
 2 [avsmax98927    ]: avs_max98927 - avs_max98927
                      Google-Eve-1.0
 3 [avsrt5663      ]: avs_rt5663 - avs_rt5663
                      Google-Eve-1.0
```
[pipewire-0.log](https://github.com/jmontleon/pixelbook-fedora/files/11715433/pipewire-0.log)
[pipewire-1.log](https://github.com/jmontleon/pixelbook-fedora/files/11715434/pipewire-1.log)
[pipewire-2.log](https://github.com/jmontleon/pixelbook-fedora/files/11715435/pipewire-2.log)
[pipewire-3.log](https://github.com/jmontleon/pixelbook-fedora/files/11715436/pipewire-3.log)

**Comment by WeirdTreeThing** [comment]
I just pushed a fix for the rt5663 UCM. As for the other ones, you need to get UCM for [max98927](https://github.com/WeirdTreeThing/chromebook-ucm-conf/tree/main/avs/avs_max98927) and [hdaudio](https://github.com/WeirdTreeThing/chromebook-ucm-conf/tree/main/avs/hdaudioB0D2). You will need to remove line 15 from HiFi.conf in avs_max98927, that line is for DSM and will not work with peakvol.

**Comment by jmontleon** [comment]
I added the other UCM configs (with line 15 commented) and update the rt5663 UCM file. I still haven't had any luck getting sound from speakers or headphones.

I don't know if these will reveal anything, but on the off chance they can I attached them.

[pipewire-card-0-try-2.log](https://github.com/jmontleon/pixelbook-fedora/files/11716164/pipewire-card-0-try-2.log)
[pipewire-card-1-try-2.log](https://github.com/jmontleon/pixelbook-fedora/files/11716165/pipewire-card-1-try-2.log)
[pipewire-card-2-try-2.log](https://github.com/jmontleon/pixelbook-fedora/files/11716166/pipewire-card-2-try-2.log)
[pipewire-card-3-try-2.log](https://github.com/jmontleon/pixelbook-fedora/files/11716167/pipewire-card-3-try-2.log)

**Comment by WeirdTreeThing** [comment]
For rt5663, I pushed a new update. For max98927, there should be a mixer control called `ssp0p_fe DSP Volume`, this is for peakvol. By default it is set to 0 which is why you don't hear anything. You need to go into alsamixer and turn it up until you can hear something, make sure it doesn't get too loud, I don't think it can but I don't have any hardware myself to test on. Once you get it working well, send an amixer contents dump for max98927.

**Comment by jmontleon** [comment]
[contents1-avsmax98927.txt](https://github.com/jmontleon/pixelbook-fedora/files/11717109/contents1-avsmax98927.txt)
[dmesg.txt](https://github.com/jmontleon/pixelbook-fedora/files/11717110/dmesg.txt)

Adjusting `ssp0p_fe DSP ` did the trick for the speakers. Turning it up to 100 didn't seem excessive to me

White noise is still the best I can get out of the headphones messing with the settings. I ran `dmesg -w` and when I selected the headphones I saw these repeating:
```
[  144.865873] snd_soc_avs 0000:00:1f.3: no matching variant found
[  144.865880] snd_soc_avs 0000:00:1f.3: create path failed: -2
[  144.865883] snd_soc_avs 0000:00:1f.3: ASoC: error at snd_soc_dai_hw_params on SSP1 Pin: -2
[  144.865887]  SSP1-Codec: ASoC: error at __soc_pcm_hw_params on SSP1-Codec: -2
[  144.865891]  Headset: ASoC: error at dpcm_fe_dai_hw_params on Headset: -2
```

Are they safe to ignore or do they indicate a problem?

**Comment by WeirdTreeThing** [comment]
> Adjusting ssp0p_fe DSP did the trick for the speakers. Turning it up to 100 didn't seem excessive to me

Thanks, I added it to the UCM, you'll need to get the new version from [here](https://github.com/WeirdTreeThing/chromebook-ucm-conf/tree/peakvol/avs/avs_max98927).

> Are they safe to ignore or do they indicate a problem?

That does indicate a problem, can you dump your NHLT table with `cp /sys/firmware/acpi/tables/NHLT ~/NHLT.bin.txt` (.txt so github will let you upload it)

**Comment by jmontleon** [comment]
Please let me know if you need anything else and I'll gather it up:
[journal.log](https://github.com/jmontleon/pixelbook-fedora/files/11723437/journal.log)
[NHLT.bin.txt](https://github.com/jmontleon/pixelbook-fedora/files/11723439/NHLT.bin.txt)

**Comment by WeirdTreeThing** [comment]
Looks like a kernel issue, try building with 6.2.8 kernel and that same patch. If that doesn't work then the patch needs to be fixed.

**Comment by jmontleon** [comment]
6.2.8 has a null pointer dereference happening. I tried 6.3.7 and it was the same as 6.4.0-rc5. I tried a 6.3.7 debug kernel but didnt notice anything more, though it was honestly kind of hard to tell. https://bugzilla.redhat.com/show_bug.cgi?id=2188969 was choking the system and causing logs to be dropped; if I build another debug kernel, I'll definitely disable that option.

[dmesg-6.2.8.txt](https://github.com/jmontleon/pixelbook-fedora/files/11728668/dmesg-6.2.8.txt)
[dmesg-6.3.7.txt](https://github.com/jmontleon/pixelbook-fedora/files/11728669/dmesg-6.3.7.txt)

I can try a later 6.2 kernel or apply a patch if you know where that was fixed. Or work on a bisect to find out if it helps.

**Comment by WeirdTreeThing** [comment]
I'm not entirely sure where/when the issue first started, I'll have to ask intel.

**Comment by WeirdTreeThing** [comment]
The issue with rt5663 is the exact same one I have had with da7219. I'm going to do some testing of my own in the next few days and see what I can find out.

**Comment by jmontleon** [comment]
Sounds good. I saw you had commented on https://github.com/thesofproject/avs-topology-xml/issues/5 when I was searching around looking at the dmesg.

Please feel free to let me know if I can gather any info or try anything out. It's really no issue building kernels with patches, whether to try things or collect more info. 

I'll keep messing with it and see if I can tease out any new information that might help.

**Comment by WeirdTreeThing** [comment]
I tested 6.3.7 on my snappy and da7219 works, so rt5663 might work on that too.

**Comment by jmontleon** [comment]
I mentioned [I tried 6.3.7](https://github.com/jmontleon/pixelbook-fedora/issues/67#issuecomment-1588172040), it behaved lie 6.4.0-rc5.

**Comment by WeirdTreeThing** [comment]
The patch may need to be fixed then.

**Comment by WeirdTreeThing** [comment]
Try this patch [ASoC-Intel-avs-Add-support-for-RT5663-codec.diff.txt](https://github.com/jmontleon/pixelbook-fedora/files/11779096/ASoC-Intel-avs-Add-support-for-RT5663-codec.diff.txt)

**Comment by jmontleon** [comment]
Yes! That works! I played with the settings some. I didn't know a lot of difference with the exception of two.

![image](https://github.com/jmontleon/pixelbook-fedora/assets/1850500/f1159559-2f95-4639-9ca4-8967a29911f4)
The highlighted STO1 causes a weird buzzing that drowns out the audio.
Turning on the Headset Mic produces an absolutely ear piercing high pitched noise. I didn't have them in, just near my ears, and it still very was unpleasant.

[rt5663-contents.txt](https://github.com/jmontleon/pixelbook-fedora/files/11782254/rt5663-contents.txt)

**Comment by jmontleon** [comment]
I built against 6.3.8, if it matters.

**Comment by WeirdTreeThing** [comment]
Ah nice, thanks for testing that patch. So you want `STO1 ADC MIXL ADC2 Switch` to be off?

**Comment by WeirdTreeThing** [comment]
UCM has been updated.

**Comment by jmontleon** [comment]
Correct,  `STO1 ADC MIXL ADC2 Switch` off.

**Comment by WeirdTreeThing** [comment]
It should be off now.

**Comment by jmontleon** [comment]
If anyone else comes along and wants to give this a try I've built a kernel along with a package to put all the configs and topology files in place. Install, reboot, and should more or less just work.

https://copr.fedorainfracloud.org/coprs/jmontleon/pixelbook-test/builds/

**Comment by lyncolnmd** [comment]
Hey. I'd like to test this build. I assume this is the precursor to native linux support for audio on the pixelbook? I am open to installing this build to test it and help improving it in any way I can. Right now, I've just used the files from the CrOS recovery and applied them according to how youve done your work here. Ive also applied your fix for Pipewire. Would you say this configuration is more seamless? the singular issue I have with my current setup is that the HDMI devices are shown regardless of being connected so when I remove my monitor it stays with the HDMI device as default. I'm hoping this fix might be more smooth

**Comment by jmontleon** [comment]
@lyncolnmd are you using Fedora or a different distro? I think my answer for what I can easily do to help (for instance build an updated kernel package and double check that an updated kernel is working as expected) and what you need to do to test will differ based on the answer to that.

**Comment by jmontleon** [comment]
I actually don't even remember if a custom kernel was required. I don't think so; just some configs/firmware so might be as simple as testing again to make sure things work.

**Comment by WeirdTreeThing** [comment]
Just saw this issue pop up again so I figured I'd give you some updates. 1: It turns out that automatic switching between different alsa cards works with UCM, so no need for acpid scripts anymore on both AVS and the legacy SKL driver ([see here for more info](https://github.com/WeirdTreeThing/chromebook-ucm-conf/commit/075b9d36e438a649b25f630e7ab8c85496fde11a)). 2: An intel audio engineer sent a patch to upstream linux that adds support for rt5663, but it isn't merged in any stable release yet AFAIK, so you will need to apply it from [here](https://patchwork.kernel.org/project/alsa-devel/patch/20230629112449.1755928-6-cezary.rojewski@intel.com/) and [here](https://patchwork.kernel.org/project/alsa-devel/patch/20230629112449.1755928-7-cezary.rojewski@intel.com/).

**Comment by lyncolnmd** [comment]
@jmontleon I am on Fedora at the moment. What would I need to do? I figured the more hands on deck the better to test. I can also test another distro if necessary. I usually bounce between arch/fedora/ubuntu with a preference for fedora
@WeirdTreeThing its great that automatic switching works and its also great that the support for rt5663 is coming. It would really be nice to have everything working as it should. I often use HDMI Audio Out with my monitor so switching cards (which does not work for me currently) can be a pain.

**Comment by WeirdTreeThing** [comment]
Quoting @jmontleon:
> I actually don't even remember if a custom kernel was required. I don't think so; just some configs/firmware so might be as simple as testing again to make sure things work.

You do need a custom kernel. Your build from around a month ago is a little old so you might want to rebuild it. The latest stable release (6.4.11) doesn't include the rt5663 patches, so you will need to backport them, links are on my message right below yours.

@lyncolnmd :
> What would I need to do?

A custom kernel is needed along with configuration files and firmware. @jmontleon had built packages about a month ago but they are outdated and missing improvements. I'd wait for him to update them first before trying AVS.

> I can also test another distro if necessary.

Fedora will work just fine.

> I often use HDMI Audio Out with my monitor so switching cards (which does not work for me currently) can be a pain.

HDMI works just fine with AVS.

**Comment by jmontleon** [comment]
@lyncolnmd I'll try to get an updated RPM and firmware/config package built before the weekend is out so you can give it a try. Will update when they're up.

**Comment by jmontleon** [comment]
@lyncolnmd there is an updated kernel building now.
https://copr.fedorainfracloud.org/coprs/jmontleon/pixelbook-test/build/6328184/

I just pulled the latest successful build out of koji and applied the rt5663 patches linked above. I built locally and tested successfully. If all goes well it should be done in a few hours.

When the build finishes you can run:
`sudo dnf copr enable jmontleon/pixelbook-test`

From there update to pick up the new kernel and install pixelbook-avs. You can (un)comment the two lines in /etc/modprobe.d/avs.conf to control whether you use AVS or not.

Right now everything shows up as 'Built-in Audio Pro`. Second down is speakers for me, above is Microphone, and below are the HDMI/DP outputs.
![image](https://github.com/jmontleon/pixelbook-fedora/assets/1850500/03c56b6e-a936-47c6-b767-5709b691374d)

**Comment by WeirdTreeThing** [comment]
Those should have proper names like Speakers and Headphones, are you missing the UCM?

**Comment by WeirdTreeThing** [comment]
In the audio settings program(pavucontrol) you can select a profile for each card, they should all say "Default".

**Comment by jmontleon** [comment]
Ah, yes, changing the profiles fixes it. The 'Pro Audio' were selected for some reason (I don't think I did this), but choosing 'Default' does the trick.

**Comment by lyncolnmd** [comment]
Does this mean we can automatically switch between sound outputs? (for example, not having to select headphones/headset when you plug one in because the audio automatically reroutes). Will this show HDMI out devices all the time or only when one is connected?

**Comment by WeirdTreeThing** [comment]
The outputs should only show up when plugged in.

**Comment by WeirdTreeThing** [comment]
@jmontleon great news, an intel dev created a patch for AVS rt5514, so eve will now be fully supported with the AVS driver! Here's the patch series if you want to try it out yourself: https://patchwork.kernel.org/project/alsa-devel/list/?series=797110&state=%2A&archive=both.

**Comment by jmontleon** [comment]
Definitely! I will give this a try (probably tomorrow), and respond back once I have a chance to try it out.

**Comment by jmontleon** [comment]
It looks like this is already applied in the 6.7 test kernels being built in Koji for Fedora.
https://koji.fedoraproject.org/koji/buildinfo?buildID=2315872

The module is not enabled yet so I'm doing a build with that changed so I can test. Should be done in just a bit.

Assuming I have success I can submit a PR to https://gitlab.com/cki-project/kernel-ark so it's enabled in the stock Fedora kernels going forward. That's easy enough to do; I've done it before to get all the other modules for the Pixelbook built.

**Comment by WeirdTreeThing** [comment]
One thing I forgot about is that there isn't a topology for rt5514 yet. I'd recommend creating an issue on the [avs topology repo](https://github.com/thesofproject/avs-topology-xml) asking if they can create one.

**Comment by jmontleon** [comment]
Request for the topology has been created.

PR to enable the module in Fedora going forward here:
https://gitlab.com/cki-project/kernel-ark/-/merge_requests/2811

**Comment by jmontleon** [comment]
Build with module enabled: https://copr.fedorainfracloud.org/coprs/jmontleon/pixelbook-test/build/6600576/

**Comment by WeirdTreeThing** [comment]
While intel never replied to your issue, they quietly were at work. They merged every topology into the main branch, added rt5514, created a makefile, and made a release. You can now get the topology needed at https://github.com/thesofproject/avs-topology-xml/releases.

**Comment by grimaldello** [comment]
Hi, first of all thank you for you hard work in order to make Fedora Linux working on the Pixelbook :)

Currently I'm using another distro on my Pixelbook, with with almost everything working properly, but I'd like to switch to Fedora.

It's not clear for me, if from now and on, all Fedora distro release (for example the latest version 40) will include this fix for sound and it's working of the box or is needed to do something in order to make sound work.

So, my question is: if I install new Fedora 40 Workstation or KDE Spin in my Pixelbook, sound is working fine without any changes needed?
If not could you please tell me what is needed to do?

Thank you.

**Comment by jmontleon** [comment]
Audio is working today. The ansible playbook in the repo will take care of downloading all the firmware files and what not if you don't want to do it manually:
https://github.com/jmontleon/pixelbook-fedora/blob/main/ansible/playbook.yml#L139

It looks like there is one newer issue, but as things go it is pretty minor:
https://github.com/jmontleon/pixelbook-fedora/issues/73
 
AVS is newer and _pretty_ close to just working. I was testing and spotted an issue with compressed topology files. None of this is distro specific, other than whether the distro compresses topology files or not. Fedora does so it is affected.
https://github.com/thesofproject/avs-topology-xml/issues/22

There is a fix. I  saw an e-mail thread today that this caused a regression for some. Fortunately the solution for the regression seems to be sorted.
https://lore.kernel.org/linux-sound/171812236450.201359.3019210915105428447.b4-ty@kernel.org/T/#t

Anyway, aside from that all that is missing is the UCM files.
https://github.com/alsa-project/alsa-ucm-conf/pull/418

Once that is merged it will eventually make it's way into Fedora. For now the `avs_max98927`, `avs_rt5514`, `avs_rt5663`, and `hdaudioB0D2` directories need to be put in place under `/usr/share/alsa/ucm2/Intel/avs` (the avs dir does not exist yet) if I am not mistaken.

And then just add a `/etc/modprobe.d/avs.conf` with
```
options snd-intel-dspcfg dsp_driver=4
options snd-soc-avs ignore_fw_version=1
```

**Comment by grimaldello** [comment]
@jmontleon  thank you for you detailed answer.

I'm going to give it a try this weekend to install Fedora!

**Comment by saxa** [comment]
Woow, I need to get into this issue deeper, lots of stuff going on here,thanks to all of you for this info and hard work.

**Comment by saxa** [comment]
Ok,I have to say that cloning the WeirdTreeThing https://github.com/WeirdTreeThing/chromebook-linux-audio repo and running setup-audio.py as root, set up my sound and its working.
